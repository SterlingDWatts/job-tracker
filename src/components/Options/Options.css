.Options {
  padding: 8px;
  font-size: 16px;
}

select.Options {
  background-color: #21b0bf;
  border: none;
  border-radius: 4px;
  outline: none;
  color: white;
}

.Options option {
  background-color: white;
  color: black;
}
