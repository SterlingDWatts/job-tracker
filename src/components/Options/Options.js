import React, { useContext } from "react";
import { useParams } from "react-router-dom";
import ApplicationsContext from "../../contexts/ApplicationsContext/ApplicationsContext";
import "./Options.css";

export default function Options(props) {
  const context = useContext(ApplicationsContext);
  const { addStatus } = context;
  const { options, selected } = props;
  const { appId } = useParams();

  function handleChange(appId, e) {
    const { value } = e.target;
    addStatus(appId, value)
  }

  const optionsComponents = options.map((option, idx) => {
    return (
      <option value={option.id} key={idx}> 
        {option.name}
      </option>
    );
  });
  return (
    <select
      className="Options"
      value={selected}
      onChange={(e) => handleChange(appId, e)}
    >
      {optionsComponents}
    </select>
  );
}
