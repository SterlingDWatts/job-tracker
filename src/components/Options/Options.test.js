import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import Options from "./Options";

describe("Options Component", () => {
  const columns = [
    [
      { id: 0, name: "Applied" },
      { id: 1, name: "Intro Interview" },
      { id: 2, name: "Recruiter Call" },
      { id: 3, name: "Sent Follow-up" },
      { id: 4, name: "Coding Challenge" },
      { id: 5, name: "Sent Follow-up" },
      { id: 6, name: "Technical Interview" },
      { id: 7, name: "Sent Follow-up" },
      { id: 8, name: "Onsite Interview" },
      { id: 9, name: "Offer Letter Date" },
    ],
  ];
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <MemoryRouter initialEntries={[{ pathname: "/applications/4" }]}>
          <Options options={columns} selected={4} />
        </MemoryRouter>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(
      <MemoryRouter
        initialEntries={[{ pathname: "/applications/4", key: "1" }]}
      >
        <Options options={columns} selected={4} />
      </MemoryRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
