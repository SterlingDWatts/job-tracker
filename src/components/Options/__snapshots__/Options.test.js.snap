// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Component renders the UI as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "1",
          "pathname": "/applications/4",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "1",
        "pathname": "/applications/4",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Options
    options={
      Array [
        Array [
          Object {
            "id": 0,
            "name": "Applied",
          },
          Object {
            "id": 1,
            "name": "Intro Interview",
          },
          Object {
            "id": 2,
            "name": "Recruiter Call",
          },
          Object {
            "id": 3,
            "name": "Sent Follow-up",
          },
          Object {
            "id": 4,
            "name": "Coding Challenge",
          },
          Object {
            "id": 5,
            "name": "Sent Follow-up",
          },
          Object {
            "id": 6,
            "name": "Technical Interview",
          },
          Object {
            "id": 7,
            "name": "Sent Follow-up",
          },
          Object {
            "id": 8,
            "name": "Onsite Interview",
          },
          Object {
            "id": 9,
            "name": "Offer Letter Date",
          },
        ],
      ]
    }
    selected={4}
  />
</Router>
`;
